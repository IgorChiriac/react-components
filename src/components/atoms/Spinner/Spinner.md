### Summary

Use `<Spinner />` whenever we need to notify the user that something is loading.

⚠️ &nbsp; We're working with the design team to settle on a single variation. When using it please don't customise it and stick to the styles and colors shown in the example.

### Example

```ts
import { colors, Spinner } from '@zopauk/react-components';

<Spinner size="100px" borderWidth="16px" />;
```
