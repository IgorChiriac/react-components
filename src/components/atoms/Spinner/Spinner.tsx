import React from 'react';
import styled, { keyframes } from 'styled-components';
import { colors } from '../../../constants/colors';

export interface ISpinnerProps {
  /**
   * Background color (3 quarters of the spinner)
   */
  backgroundColor?: string;

  /**
   * Width of the border
   */
  borderWidth?: string;

  /**
   * Front color (1 quarter of the spinner)
   */
  frontColor?: string;

  /**
   * Size of the spinner
   */
  size?: string;
}

const spin = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

const SSpinner = styled.div<ISpinnerProps>`
  width: ${({ size = '45px' }) => size};
  height: ${({ size = '45px' }) => size};
  margin: 8px;
  border: ${({ borderWidth = '8px' }) => borderWidth} solid ${({ frontColor = colors.base.primary }) => frontColor};
  border-radius: 50%;
  border-top-color: ${({ backgroundColor = colors.neutral.light }) => backgroundColor};
  animation: ${spin} 1.2s linear infinite;
`;

const Spinner: React.FC<ISpinnerProps> = props => <SSpinner {...props} />;

export default Spinner;
