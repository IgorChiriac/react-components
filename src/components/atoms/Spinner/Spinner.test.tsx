import React from 'react';
import { render } from '@testing-library/react';
import Spinner from './Spinner';

describe('<Spinner />', () => {
  it('renders the component', () => {
    const { container } = render(<Spinner />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the component with props', () => {
    const { container } = render(<Spinner size="38px" borderWidth="4px" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
