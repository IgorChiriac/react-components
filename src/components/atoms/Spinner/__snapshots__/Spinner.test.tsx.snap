// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> renders the component 1`] = `
.c0 {
  width: 45px;
  height: 45px;
  margin: 8px;
  border: 8px solid #00D9C5;
  border-radius: 50%;
  border-top-color: #EFEFEF;
  -webkit-animation: iVXCSc 1.2s linear infinite;
  animation: iVXCSc 1.2s linear infinite;
}

<div
  class="c0"
/>
`;

exports[`<Spinner /> renders the component with props 1`] = `
.c0 {
  width: 38px;
  height: 38px;
  margin: 8px;
  border: 4px solid #00D9C5;
  border-radius: 50%;
  border-top-color: #EFEFEF;
  -webkit-animation: iVXCSc 1.2s linear infinite;
  animation: iVXCSc 1.2s linear infinite;
}

<div
  class="c0"
/>
`;
